### PR TITLE
use assert(0) instead of null reference

### PR DESCRIPTION
**Pull request by WalterBright**
as that's what `assert(0)` is for.

### CODE REVIEW

**Comment by yebblies** [comment]
Can we get rid of halt?  Otherwise why bother changing all those calls...?

**Comment by WalterBright** [comment]
> Can we get rid of halt?

The dmd C++ source code doesn't have access to D's `assert(0)`. So, no.

**Comment by yebblies** [comment]
If the C++ side is the only user then it can be moved out of the D code.

**Comment by WalterBright** [comment]
I prefer the assert(0) rather than the hackish C++ version.

File: src/errors.d [hunk 1]
```diff
@@ -286,8 +286,5 @@ extern (C++) void fatal()
  */
 extern (C++) void halt()
```
Comment (andralex) [inline src/errors.d]: so then how about removing halt's definition

**Comment by andralex** [comment]
assert(0) looks cleaner to me too

**Pull request merged**